salt/fileserver/roots.py

```diff
@@ -94,6 +95,7 @@ def _add_file_stat(fnd):
             return _add_file_stat(fnd)
         return fnd
     for root in __opts__["file_roots"][saltenv]:
+        root = root.replace("__env__", actual_saltenv)
```

Comment:
Would it make more sense to use something like this?
```suggestion
        if root == "__env__":
            root = actual_saltenv
```
That way it won't replace `__env__` in other environments where users may be using that as a prefix/something similar.

Reply:
I kinda like the flexibility of being able to replace for any env, but I definitely get your point. I don't want to break anyone. I'll see about that change.

Reply:
@MKLeb , I changed the logic to only substitute `__env__` found in the root paths of the `__env__` environment.